### PR TITLE
📺 Add completions for fish shell

### PR DESCRIPTION
**Pull request by ylectric**
This changeset adds completions for [fish shell.](https://fishshell.com)
[asciicast](https://asciinema.org/a/224252)

Once this PR is approved, I would be happy to submit another one to also support fish in Homebrew formula.

### CODE REVIEW

**Comment by phatblat** [comment]
@ylectric this is awesome! 🎉  I'm a fish shell user as well and this has been on my TODO list but I haven't gotten to it. Thanks!

This repo actually has a copy of the mas homebrew formula so that things like this can get added as part of a normal PR. Eventually, I hope to automate the formula updates from these source files:

- [`mas.rb`](https://github.com/mas-cli/mas/blob/master/Homebrew/mas.rb) - a copy of our [core formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/mas.rb)
- [`mas-tap.rb`](https://github.com/mas-cli/mas/blob/master/Homebrew/mas-tap.rb) - a copy of our [custom tap formula](https://github.com/mas-cli/homebrew-tap/blob/master/Formula/mas.rb) for older macOS support

Each of those files needs a [`fish_completion`](https://www.rubydoc.info/github/Homebrew/brew/Formula:fish_completion) line if you would be so kind as to add them 🙏 .

I would think this file would need to get copied into `~/.config/fish/completions`, but that appears to be for user-created files.

```
$ ls -l ~/.config/fish/completions
total 8
-rw-r--r--  1 ben                 18 Nov  4 09:38 fisher.fish
```

However, the [fish docs](http://fishshell.com/docs/current/index.html#completion-path) say programs including completions should put them in the "vendor" dir which is system dependent.

```
$ pkg-config --variable completionsdir fish
/usr/local/share/fish/vendor_completions.d

$ ls -l /usr/local/share/fish/vendor_completions.d
total 0
lrwxr-xr-x  1 ben                 77 Nov 17 18:32 carthage.fish -> ../../../Cellar/carthage/0.31.2/share/fish/vendor_completions.d/carthage.fish
lrwxr-xr-x  1 ben                 66 Jan 29 21:42 hub.fish -> ../../../Cellar/hub/2.8.4/share/fish/vendor_completions.d/hub.fish
```

## TL;DR

Turns out that Homebrew already takes care of this (see [carthage](https://github.com/Homebrew/homebrew-core/blob/master/Formula/carthage.rb#L22)).

**Comment by phatblat** [comment]
this. is. amazing. 😍 
screen shot 2019-01-29 at 10 08 31 pm

**Comment by ylectric** [comment]
@phatblat Sorry, my bad, I didn't realise that formulae copies reside in this repo as well.

Homebrew ensures that completions are put into appropriate locations, so there's no fuss for us.
I will amend the formulae and submit an update.

I see `--dry-run` on your screenshot, does it mean that `mas` actually has flags? If so, is there any way to list them besides groking the sources? Also, I used a few lines of Python to generate fish completions, but I'm not familiar enough with XCode build systems to evaluate what would it take to make this process fully automated. What do you think?

**Comment by ylectric** [comment]
@phatblat Thank you for a thorough explanation. It should be enough to get the very first version of completions, and I may think about auto-generation later.

**Comment by ylectric** [comment]
Alright, I amended Homebrew formulae and made completions for `uninstall` a little bit smarter.
[asciicast](https://asciinema.org/a/mHhxOQTF7ox1YlE4y8vBw6k0k)

**Comment by ylectric** [comment]
Jenkins failure looks unrelated, could you double check?
```
22:07:32 ==> 👢 Bootstrapping
22:07:32 /usr/local/opt/ruby/bin/bundle:23:in `load': cannot load such file -- /usr/local/lib/ruby/gems/2.6.0/gems/bundler-2.0.1/exe/bundle (LoadError)
22:07:32 	from /usr/local/opt/ruby/bin/bundle:23:in `<main>'
```

**Comment by phatblat** [comment]
Sorry! That was a temporary system issue on my build server. It's fixed now.

I love the fancy uninstall completion. Is this PR ready to go now?

**Comment by ylectric** [comment]
@phatblat yes, absolutely :)
